client/javascript-esm/packages/jolokia/src/jolokia-utils.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  JolokiaErrorResponse,
+  JolokiaSuccessResponse,
+  VersionResponseValue
+} from "./jolokia-types.js"
+
+export type ParseResult<T> = { hasError: false; parsed: T } | { hasError: true; error: string }
+
+function isObject(value: unknown): value is object {
+  const type = typeof value
+  return value != null && (type === 'object' || type === 'function')
+}
+
+export function isResponseSuccessType(o: unknown): o is JolokiaSuccessResponse {
+  return isObject(o) && 'status' in o && 'timestamp' in o && 'value' in o
+}
+
+export function isResponseErrorType(o: unknown): o is JolokiaErrorResponse {
```

Comment:
So we already have `isError(resp: JolokiaResponse)`. The only difference and its value proposition should be the type of the argument, where the original is `JolokiaReponse` while the new function is `unknown`.

We can also shorten the name as it's a bit redundant. Maybe `isResponseError(o: unknown)`?